### ISSUE TITLE
Proposal: platform startup sequence

### ISSUE THREAD

**Issue by cody-littley** [hashgraph/hedera-services#5853]
TODO generate documentation on our planned startup sequence.

**Comment by cody-littley** [comment]
@lpetrovic05 @poulok This is the way I think we want platform startup to work after the current round of refactors is concluded. This is basically the same has we have discussed, I just wanted to formalize it a bit more.

**Comment by poulok** [comment]
Why are some blocks next to each other? Does that mean it doesn't matter which order those happen in?

**Comment by poulok** [comment]
It would also be helpful to have some text that goes along with the diagram explaining which steps have a hard dependency on previous steps, or make assumptions that the system is in some state. I think some of these exist for `SwirldState.init()` and `SwirldMain.init()`.

Does it also make sense to show the genesis startup sequence here?

**Comment by cody-littley** [comment]
@poulok 

> It would also be helpful to have some text that goes along with the diagram explaining which steps have a hard dependency on previous steps

This is actually a hard thing to represent in this diagram... or at least, a hard thing to do across the board. Some steps in the startup sequence depend on almost everything that come before. If I added that to the diagram it would get very cluttered.

> Does it also make sense to show the genesis startup sequence here?

The latest iteration of the diagram has this.

> Why are some blocks next to each other? Does that mean it doesn't matter which order those happen in?

The latest iteration of the diagram no longer includes this.

**Comment by poulok** [comment]
> This is actually a hard thing to represent in this diagram

Yes, that's why I think some text should accompany the diagram. Diagrams are almost never enough by themselves.
